hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -243,6 +248,15 @@ def check_valid_minmax(prefix, val, info_obj):
                 check_valid_interval(min_value, max_value, "min_value", "max_value")
             kw["max_value"] = max_value
 
+        # By default, floats() will generate subnormals if they are in the
+        # inferred values range. If we have detected that xp flushes to zero for
+        # the passed dtype, we ensure floats() will not generate subnormals
+        # unless allow_subnormal=True is passed.
+        if allow_subnormal is not None:
+            kw["allow_subnormal"] = allow_subnormal
+        elif widths_ftz[finfo.bits]:
+            kw["allow_subnormal"] = False
```

Comment:
I think it'll be cleaner if we do our `widths_ftz` check here, instead of calculating global state and passing around another argument - it's compact and reasonably fast code.

```suggestion
        else:
            finfo = xp.finfo(dtype)
            subnormal = next_down(finfo.smallest_normal, width=finfo.bits)
            try:
                if xp.asarray(subnormal, dtype=dtype) == 0:
                    kw["allow_subnormal"] = False
            except Exception:
                pass
```

Reply:
Yep this makes perfect sense. Removed the global state thing now but will let you resolve this.